### ISSUE TITLE
Script Settings Custimization

### ISSUE THREAD

**Issue by garibaldi0** [PresidioCode/SecureCRT#23]
I've been looking into a way to not overwrite my customized settings with each update and wanted to start a discussion here.

The current import of script_settings.py works, but is there a better option.  Maybe using ConfigParser, or a JSON or YAML config, etc. etc.

If we stick with the current import option.  I'd suggest we move the default settings to say script_settings_default.py and use the following code to import.

``` python
# Import Settings from Settings File or Default settings
try:
    from script_settings import settings
except IOError:
    from script_settings_default import settings
```

This way, if I want to change the settings, I create a script_settings.py with _all_ settings and it wouldn't get over written the next time I update.

Thoughts?

**Comment by jamiecaesar** [comment]
My only goal was that the settings file was easy to edit manually in a text editor.   Since it only contained a few variables, it seemed easiest (at the time, at least) to just make it a python file that could be imported.  I'm not against changing it to another format, as long as there is a benefit to gain from it.

I definitely agree that we need some way to keep people's settings from getting overwritten.   Assuming we work with how it is now, I like your idea of having default settings that can be updated while not touching everyone's individual settings.   We could even add the settings filename to .gitignore so it won't even try to add that to any pushes.  When someone clones the repo, they won't have a settings file, and so it will get written from the defaults when the first script runs.

I think I'd like to take your idea a little further and instead of just reading the default settings if the personalized ones don't exist, create the personalized settings file if it is missing.  It could be as simply as copying the default settings file, or maybe if we used a different format.

Maybe we could get really crazy and track a version number in there or something, so when your settings file isn't the latest version it will give a pop-up that you need to update it (and maybe even have a script that will update the settings file for you -- read the old values, write a new file but putting the original values back unless they didn't exist -- then put in the default value).    Or maybe it is just easier that if a newer script can't find the setting it needs, it will throw a pop-up error and recommend refreshing the settings file.

Anyway, in short, I think it is a good idea and there are probably a lot of ways we could approach fixing the issue.

**Comment by jamiecaesar** [comment]
I've push an update to dev that works on my Mac.   I changed the behavior so if the custom settings don't exist, then copy the default settings to the customer settings file, and pull from that.   That way, we don't run into issues with people modifying defaults and having the same problem down the road.

I'll test this on the windows machine later, but if you want to give it a try also that would be helpful.   Assuming it all works well, I'll merge this into the master branch.

One thing that I've noticed (or was reminded of while playing with this) is that when dealing with imports on SecureCRT, you have to close and re-open SecureCRT to clear out the imports.   Otherwise, you'll try something once, modify the code and then it has the exact same behavior as before the modifications.   You have to restart SecureCRT to get the new code to work.

**Comment by jamiecaesar** [comment]
In the re-write branch I've changed to using a JSON file for settings (and ignore .json files in the .gitignore) so that settings won't be touched when scripts are updated.
